apps/rush-lib/src/cli/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -6,6 +6,15 @@ import * as semver from 'semver';
 import Utilities from '../../../utilities/Utilities';
 import { BaseShrinkwrapFile } from '../base/BaseShrinkwrapFile';
 
+// This is based on PNPM's own configuration:
+// https://github.com/pnpm/pnpm-shrinkwrap/blob/master/src/write.ts
+const SHRINKWRAP_YAML_FORMAT: yaml.DumpOptions = {
+  lineWidth: 1000,
+  noCompatMode: true,
+  noRefs: true,
+  sortKeys: true
+};
```

Comment:
@zkochan do you think this configuration should be exported by the [pnpm-shrinkwrap](https://github.com/pnpm/pnpm-shrinkwrap/) library?  Could that library help with the operations in this file?

Reply:
can't you just use one of the write functions from `pnpm-shrinkwrap`?

Reply:
-msft said that he looked at this library, but the API was unclear. For example, you have:
> **write(pkgPath, wantedShrinkwrap, currentShrinkwrap) => Promise<void>**
> Writes the wanted/current shrinkwrap files. When they are empty, removes them.

What are the "wanted" and "current" files?  The data type for `wantedShrinkwrap` is a `Shrinkwrap` interface, which seems useful, but it appears to be a forgotten export for this API.

This one looks useful:

> **writeCurrentOnly(pkgPath, currentShrinkwrap) => Promise<void>**
> Writes the current shrinkwrap file only. Fails if there is no node_modules directory in the pkgPath

...but it says that it expects a **node_modules** folder for some reason.  Why?  Seems like an arbitrary requirements. 

Lastly, it might be nice to have non-async versions of these functions.  We often avoid async in our tools because it makes the control follow more difficult to follow and debug.  And the single-threaded parallelism tends to be not provide much speed improvements, because (1) NodeJS makes it difficult to bound the parallelism, and too many async operations can actually slow things down by causing too much context switching or disk thrashing, and (2) Rush already launches a separate node.exe process for each CPU core, which is perhaps a better behaved form of parallelism.

Reply:
The wanted shrinkwrap is in the root of your project. In the same directory where the `package.json` is. The current `shrinkwrap.yaml` is the one in `node_modules/.shrinkwrap.yaml`. I don't know which one you need but pnpm writes both.

I don't have objections having sync versions or exporting the configs passed to YAML